dmutils/forms/validators.py

```diff
@@ -8,15 +8,66 @@
 
 import re
 
-from wtforms.validators import Regexp, ValidationError
+from wtforms.validators import ValidationError
+
+
+class EmailValidator:
+    # Largely copied from https://github.com/alphagov/notifications-utils/blob/\
+    #   67889886ec1476136d12e7f32787a7dbd0574cc2/notifications_utils/recipients.py
+    #
+    # regexes for use in validate_email_address.
+    # invalid local chars - whitespace, quotes and apostrophes, semicolons and colons, GBP sign
+    # Note: Normal apostrophe eg `Firstname-o'surname@domain.com` is allowed.
+    _INVALID_LOCAL_CHARS = r"\s\",;:@£“”‘’"
+    _email_regex = re.compile(r'^[^{}]+@([^.@][^@]+)$'.format(_INVALID_LOCAL_CHARS))
+    _hostname_part = re.compile(r'^(xn-|[a-z0-9]+)(-[a-z0-9]+)*$', re.IGNORECASE)
+    _tld_part = re.compile(r'^([a-z]{2,63}|xn--([a-z0-9]+-)*[a-z0-9]+)$', re.IGNORECASE)
```

Comment:
what's with the `xn`s here and for the hostname?

Reply:
Ah https://en.wikipedia.org/wiki/Internationalized_domain_name